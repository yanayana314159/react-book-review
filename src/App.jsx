import logo from "./logo.svg";
import "./App.scss";
import { Router } from "./routes/Router";
import { SignIn } from "./pages/SignIn";
import { Home } from "./pages/Home.jsx";

function App() {
  return (
    <div className="App">
      <Router />
    </div>
  );
}

export default App;
