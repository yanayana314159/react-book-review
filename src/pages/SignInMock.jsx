import React, { useState } from "react";

export const SignInMock = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const handleEmailChange = (e) => setEmail(e.target.value);
  const handlePasswordChange = (e) => setPassword(e.target.value);
  const baseurl = "https://railway.bookreview.techtrain.dev";

  return (
    <div>
      <main className="signin">
        <form className="signin-form">
          <h2>ログイン</h2>
          <label className="email-label" htmlFor="email">
            メールアドレス
          </label>
          <br />
          <input
            type="email"
            className="email-input"
            id="email"
            autoComplete="current-password"
            onChange={handleEmailChange}
            placeholder="email"
          />
          <br />
          <label className="password-label" htmlFor="pw">
            パスワード
          </label>
          <br />
          <input
            type="password"
            className="password-input"
            id="pw"
            autoComplete="current-password"
            onChange={handlePasswordChange}
            placeholder="password"
          />
          <br />
          <button type="button" className="signin-button">
            ログイン
          </button>
        </form>
      </main>
      <a>新規作成</a>
    </div>
  );
};
