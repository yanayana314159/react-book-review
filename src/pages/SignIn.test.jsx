import React from "react";
import { render, screen } from "@testing-library/react";
import { SignInMock } from "./SignInMock.jsx";

test("コンポーネントの確認", () => {
  const screen = render(<SignInMock />);
  //ボタンの確認
  const buttonElement = screen.getByRole("button", { name: "ログイン" });
  expect(buttonElement).toBeInTheDocument();
  //ラベルの確認
  expect(screen.getByLabelText("メールアドレス")).toBeInTheDocument();
  expect(screen.getByLabelText("パスワード")).toBeInTheDocument();
  //入力フォームの確認
  expect(screen.getByPlaceholderText("email"));
  expect(screen.getByPlaceholderText("password"));
});

/*
import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
*/
