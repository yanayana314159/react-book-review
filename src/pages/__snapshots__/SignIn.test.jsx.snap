// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントの確認 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <main
          class="signin"
        >
          <form
            class="signin-form"
          >
            <h2>
              ログイン
            </h2>
            <label
              class="email-label"
            >
              メールアドレス
            </label>
            <br />
            <input
              autocomplete="current-password"
              class="email-input"
              type="email"
            />
            <br />
            <label
              class="password-label"
            >
              パスワード
            </label>
            <br />
            <input
              autocomplete="current-password"
              class="password-input"
              type="password"
            />
            <br />
            <button
              class="signin-button"
              type="button"
            >
              ログイン
            </button>
          </form>
        </main>
        <a>
          新規作成
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <main
        class="signin"
      >
        <form
          class="signin-form"
        >
          <h2>
            ログイン
          </h2>
          <label
            class="email-label"
          >
            メールアドレス
          </label>
          <br />
          <input
            autocomplete="current-password"
            class="email-input"
            type="email"
          />
          <br />
          <label
            class="password-label"
          >
            パスワード
          </label>
          <br />
          <input
            autocomplete="current-password"
            class="password-input"
            type="password"
          />
          <br />
          <button
            class="signin-button"
            type="button"
          >
            ログイン
          </button>
        </form>
      </main>
      <a>
        新規作成
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
